crates/criticalup-core/src/download_server_client.rs

```diff
@@ -113,13 +120,36 @@ impl DownloadServerClient {
     fn send(&self, builder: RequestBuilder) -> Result<Response, Error> {
         let req = builder.build().expect("failed to prepare the http request");
         let url = req.url().to_string();
-        let response = self
-            .client
-            .execute(req)
-            .map_err(|e| Error::DownloadServerError {
-                kind: DownloadServerError::Network(e),
-                url,
-            })?;
+
+        // Retry logic implemented here.
+        // This section implements a quick and dirty version of a retry a failed request for
```

Comment:
This comment does some of pearl clutching and making excuses for itself. I think you can be confident and leave just the note that this is a retry (if that, even, the `current_retries` and `current_retry_backoff` should be a pretty good tip off.)